src/Core/Router.php

```diff
@@ -2,60 +2,51 @@
 
 namespace TrendingRepos\Core;
 
-class Router {
-    public $routes = [
-        'GET' => [],
-        'POST' => []
-    ];
+use TrendingRepos\Exception\RouterException;
 
-    public static function load($file) {
-        $router = new static;
-        require $file;
+class Router 
+{
+    public $routes;
+    public $request;
 
-        return $router;
+    public function __construct(array $routes, Request $request)
+    {
+        $this->routes = $routes;
+        $this->request = $request;
     }
 
-    public function get($uri, $controller) {
-        $parameters = (new Request)->parameters();
-        if(!empty($parameters)) {
-            foreach($parameters as $key => $value) {
-                $uri = trim($uri, $key . '=' . $value);
-            }
-            $uri = trim($uri, '/?');
-
-           if($uri  == null) {
-               $uri = '';
-           }
+    public function get() 
+    {
+        if (!array_key_exists($this->request->methodType(), $this->routes)) {
+            throw new RouterException('Routes file structure in not as we expect');
         }
 
-        $this->routes['GET'][$uri] = $controller;
+        if (!$this->uriExists($this->request->getSlug())) {
+            throw new RouterException('This is not the web page you are looking for!');
+        }
+        $this->direct($this->request->getSlug(), $this->request->methodType());
```

Comment:
You've got the idea, almost perfect! Just add a new line before the last call 😉

Reply:
I should add new line before any last call ? even if it's not `return` ?

Reply:
Ok got the answer from next comment :+1: